src/evented-tokenizer.ts

```diff
@@ -266,18 +266,18 @@ export default class EventedTokenizer {
     endTagName() {
       let char = this.consume();
 
-      if (isSpace(char)) {
-        this.transitionTo(TokenizerState.beforeAttributeName);
-        this.tagNameBuffer = '';
+      if (isSpace(char) && isAlpha(this.peek())) {
+        this.delegate.reportSyntaxError('closing tag must only contain tagname');
       } else if (char === '/') {
-        this.transitionTo(TokenizerState.selfClosingStartTag);
-        this.tagNameBuffer = '';
+        this.delegate.reportSyntaxError('closing tag cannot be self-closing');
       } else if (char === '>') {
         this.delegate.finishTag();
         this.transitionTo(TokenizerState.beforeData);
         this.tagNameBuffer = '';
       } else {
-        this.appendToTagName(char);
+        if (!this.delegate.current().syntaxError && !isSpace(char)) {
```

Comment:
I don’t fully understand this conditional (reviewing on mobile so forgive me if I’ve missed something obvious).

Why do we check if `.current().syntaxError`?

Reply:
You're right that this is confusing, and I feel like there may be a better way to do this.

This check is required since we no longer enter the `beforeAttributeName` or `selfClosingStartTag` states, which would reset the `tagNameBuffer`. Without this check I was getting invalid tag names that include the whitespace and/or attributes. i.e `{tagname: 'div  foo="bar"'}`.

The `!isSpace(char)` is there to not include whitespace in the tagname, since I had made the decision to allow trailing whitespace in the closing tag. More on that decision in my reply to your other comment.